### PR TITLE
Adding isOrderCancelled and isOrderCancelling cases

### LINKED ISSUE #612: Cancel order: Sync statuses in Explorer and CowSwap
The issue is related to the 'Cancel' order transactions.

When I cancel an order, it is immediately changes its status as 'Cancelled' in the  protocol explorer.
However, as we use a 'soft' cancellation, the order might be executed, hence, the status is changed to 'Filled' in the explorer, and it seems to be strange a bit. 
I think it would be nice to add 'Cancelling' status in the protocol explorer (link it is in the CowSwap app) for an order and show it until an order is successfully cancelled/filled.
https://drive.google.com/file/d/1j92oV8O2LLPMGZgCY4FSUnFZrXXhjpVL/view

### PR DESCRIPTION
**Pull request by henrypalacios**
# Summary

Closes #612 

**Proposal:**
Add logic to check if the order is cancelled, [cowswap reference](https://github.com/gnosis/cowswap/blob/2165-adding-wallet-supports/src/custom/state/orders/utils.ts#L38)

# To Test

1. Create a new swap on [dev cowswap environment](https://cowswap.dev.gnosisdev.com/#/swap).
2. Go to pending button and clicked, it will open `account modal` try to cancel order.
3. Go to explorer https://pr999--gpui.review.gnosisdev.com/rinkeby/address/<address>    (:bulb:  replace your wallet `address`)

### CODE REVIEW

**Review by elena-zh — APPROVED** [review]
Hey @henrypalacios , great changes!!
I have retested it in all 3 networks, and it works!

File: test/utils/operator/orderStatus.test.ts [hunk 1]
```diff
@@ -114,6 +114,9 @@ describe('Filled status', () => {
 })
 
 describe('Canceled status', () => {
+  const currentDate = new Date(DATE.setFullYear(DATE.getFullYear() + 4))
```
Comment (ramirotw) [inline test/utils/operator/orderStatus.test.ts:117]: clarify please what's that 4 magic number about

**Review by ramirotw — CHANGES_REQUESTED** [review]

File: test/utils/operator/orderStatus.test.ts (hunk 1 above)
Comment (henrypalacios) [reply test/utils/operator/orderStatus.test.ts]: Done!

**Review by ramirotw — APPROVED** [review]

File: src/utils/operator.ts [hunk 2]
```diff
@@ -44,15 +44,35 @@ function isOrderPresigning(order: RawOrder): boolean {
...
+/**
+ * An order is considered cancelled if the `invalidated` flag is `true` and
+ * it has been at least `PENDING_ORDERS_BUFFER` since it has been created.
+ * The buffer is used to take into account race conditions where a solver might
+ * execute a transaction after the backend changed the order status.
+ *
+ * We assume the order is not fulfilled.
+ */
+function isOrderCancelled(order: Pick<RawOrder, 'creationDate' | 'invalidated'>): boolean {
+  const PENDING_ORDERS_BUFFER = 60 * 1000 //60s
```
Comment (anxolin) [inline src/utils/operator.ts:56]: Shouldn't this be a global constant?

Comment (henrypalacios) [reply src/utils/operator.ts]: I have moved it to a global constant to make it available.

File: test/utils/operator/orderStatus.test.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ function _getPastTimestamp(): number {
 }
 
 // mockTimes set's Date.now() to DATE const in the test context
-beforeEach(mockTimes)
+beforeEach(() => mockTimes)
```
[the remark this answers is not in the document]
Comment (henrypalacios) [reply test/utils/operator/orderStatus.test.ts]: Yes, I have added a test case for the new `cancelling` state.

**Review by alfetopito — APPROVED** [review]
Perfect, working great

**Comment by alfetopito** [comment]
Since there's still time, I think this can be included in the pending 2.6.0 release

**Pull request merged**